File: attentive-android-sdk/src/main/java/com/attentive/androidsdk/internal/util/VerboseTree.java
package com.attentive.androidsdk.internal.util;

import android.annotation.SuppressLint;
import android.util.Log;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import timber.log.Timber;

public class VerboseTree extends Timber.Tree {
    @SuppressLint("LogNotTimber")
    @Override
    protected void log(int priority, @Nullable String tag, @NonNull String message, @Nullable Throwable t) {
        if (priority == Log.ERROR) {
            Log.e(tag, message, t);
        } else if (priority == Log.WARN) {
            Log.w(tag, message, t);
        } else if (priority == Log.INFO) {
            Log.i(tag, message, t);
        } else if (priority == Log.DEBUG) {
            Log.d(tag, message, t);
        } else if (priority == Log.VERBOSE) {
            Log.v(tag, message, t);
        } else if (priority == Log.ASSERT) {
            Log.wtf(tag, message, t);
        }
    }
}

File: attentive-android-sdk/src/main/java/com/attentive/androidsdk/internal/util/VerboseTree.kt
package com.attentive.androidsdk.internal.util

import android.annotation.SuppressLint
import android.util.Log
import timber.log.Timber

class VerboseTree : Timber.Tree() {
    @SuppressLint("LogNotTimber")
    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        val attentiveMessage = "Attentive: $message"
        if (priority == Log.ERROR) {
            if (t != null) Log.e(tag, attentiveMessage, t) else Log.e(tag, attentiveMessage)
        } else if (priority == Log.WARN) {
            if (t != null) Log.w(tag, attentiveMessage, t) else Log.w(tag, attentiveMessage)
        } else if (priority == Log.INFO) {
            if (t != null) Log.i(tag, attentiveMessage, t) else Log.i(tag, attentiveMessage)
        } else if (priority == Log.DEBUG) {
            if (t != null) Log.d(tag, attentiveMessage, t) else Log.d(tag, attentiveMessage)
        } else if (priority == Log.VERBOSE) {
            if (t != null) Log.v(tag, attentiveMessage, t) else Log.v(tag, attentiveMessage)
        } else if (priority == Log.ASSERT) {
            if (t != null) Log.wtf(tag, attentiveMessage, t) else Log.wtf(tag, attentiveMessage)
        }
    }
}
